### PR TITLE
some mac build instructions

### PR DESCRIPTION
**Pull request by jgrewe**
plus a cmake target to create the zip archive for macOS

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -86,6 +86,20 @@ if(DEBUG_GLUE)
    add_definitions(-DDEBUG_GLUE=1)
 endif()
 
+
+
+if(${CMAKE_SYSTEM_NAME} MATCHES "Darwin" AND ${CMAKE_VERSION} GREATER 3.2)
+  add_custom_target(macOS_zip COMMAND
+    ${CMAKE_COMMAND} -E tar "cfv" "nix_mx_macOS.zip" --format=zip
```
Comment (gicmo) [inline CMakeLists.txt]: I think it would be cool the include the version in the zip file name.

File: MacOSBuild.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
+How to build NIX-MX on MacOS
+----------------------------
+
+To compile NIX-MX for MaOS for your MATLAB Version follow these steps:
+
+**Dependencies**
+- you have to have a working build environment including a c/c++ compiler
```
Comment (gicmo) [inline MacOSBuild.md]: Start with captial y? Rmove the "c/" part before compiler. (I would also capitalize C++).

File: MacOSBuild.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+- install the nix c++ libraries (e.g. via homebrew `brew install nixio`)
+to build it from source please visit [NIX](https://github.com/G-Node/nix).
+- install cmake and the boost libraries (e.g. via homebrew `brew install boost; brew install cmake`
+
+- clone the nix-mx repository ''''git clone https://github.com/g-node/nix-mx.git''''
+- change into the nix-mx directory and create a build subfolder: `mkdir build`
+- change into the build folder and configure the build: `cd build; cmake ..`
+- compile nix-mx: `make -j`
```
Comment (gicmo) [inline MacOSBuild.md]: `-j` without argument means no limit, might create flames in the computer ;D

File: README.md [hunk 4]
```diff
@@ -33,3 +33,23 @@ NOTE: there was a small bugfix in examples/Primer.m after the latest release. If
 **Build NIX-MX under Windows**
 
 To build NIX-MX under Windows please follow the guide provided at [WinBuild.md](https://github.com/G-Node/nix-mx/blob/master/WinBuild.md)
+
+
+Getting Started (MacOS)
```
Comment (gicmo) [inline README.md]: it is macOS, with small M ;D

File: README.md [hunk 5]
```diff
@@ -33,3 +33,23 @@ NOTE: there was a small bugfix in examples/Primer.m after the latest release. If
...
+-------------------------------
+
+**Quick start packages, Beta-Release 1.4.0**
+
+The quick start packages (https://github.com/G-Node/nix-mx/releases) are compiled under MacOS Sierra using Matlab 2016b and contain the compiled mex files, nix m-Files, tests, and a stratup.m script. To run it unzip the file and run the startup.m file in MATLAB. This file simply adds the current folder containing the mex files to the MATLAB path. Do not change the file/folder structure.
```
Comment (gicmo) [inline README.md]: same (macOS)

File: CMakeLists.txt (hunk 1 above)
Comment (gicmo) [inline CMakeLists.txt]: Also, couldn't we use `zip` and have it also work on Linux?

Comment (jgrewe) [inline CMakeLists.txt]: We still have the hdf problem under linux, methinks. Hence, i limited that to mac

File: MacOSBuild.md (hunk 3 above)
Comment (jgrewe) [inline MacOSBuild.md]: hehe, I know. personally I like to hear the fan turning on :)

**Pull request merged**